ananse/network.py

```diff
@@ -623,178 +299,119 @@ def create_promoter_network(self, ddf, prom, fin_expression, outfile, column="tp
             ).mean(1),
             columns=[column],
         )
-        expression.index = [i.upper() for i in list(expression.index)]
-        # print(expression)
         expression[column] = np.log2(expression[column] + 1e-5)
 
+        expression.index.rename("target", inplace=True)
         expression = expression.reset_index()
-        expression=expression.rename(columns={"index":"gene", "tpm":"target_expression"})
-
-        network = prom_table.merge(expression, how='left', left_on="gene",right_on="gene")
-        network = network.merge(expression, how='left', left_on="factor",right_on="gene")
-        network = network.rename(columns={"gene_x":"gene", "target_expression_x":"target_expression", "target_expression_y":"factor_expression"})
-        network = network.drop(columns=["gene_y"])
-
-        # network = network.compute()
-        with ProgressBar():
-            network = network.compute(num_workers = self.ncore)
-
-        for col in ["factor_expression", "target_expression", "max_binding_in_promoter"]:
-            network[col + ".scale"] = minmax_scale(network[col])
-            network[col + ".rank.scale"] = minmax_scale(rankdata(network[col]))
-
-        exclude_cols = [
-            "factor",
-            "gene",
-            "factor_expression",
-            "target_expression",
-            "factor_expression.scale",
-            "target_expression.scale",
-            # "factor_expression.rank.scale", 
-            # "target_expression.rank.scale",
-            # "max_binding_in_promoter",
-            "max_binding_in_promoter.scale",
-            "max_binding_in_promoter.rank.scale",
-        ]
-        network = network[[c for c in network.columns if c not in exclude_cols]]
-        network = network.set_index("source_target")
+        expression = expression.rename(columns={"tpm": "target_expression"})
 
-        bp = network.mean(axis=1)
-        bpd = pd.DataFrame(bp)
-        bpd = bpd.rename(columns={0: "binding"})
-        bpd["prob"] = minmax_scale(rankdata(bpd["binding"], method="dense"))
+        if tfs is None:
+            package_dir = os.path.dirname(ananse.__file__)
+            tffile = os.path.join(package_dir, "db", "tfs.txt")
+            tfs = pd.read_csv(tffile, header=None)[0].tolist()
```

Comment:
- If `tfs` is specified, it must currently be a list, I think a file might be nicer.
 - No option to pass tfs over the CLI.
 - No option to _not_ filter by TFs

Reply:
I added this functionality for now only to use via the API. It makes generating networks for a limited set of TFs *much* faster. Later on, this would also be exposed via the CLI. 
I actually like it to be a list for this function. The CLI could actually also use a file.
Given that the point of ANANSE is predicted TF-target interactions, what would be the use case of NOT filtering? :) Actually, what it mainly does now is removing wrong IDs from the motif file that cannot be resolved.